x-pack/plugins/ml/public/register_helper/register_search_links/search_deep_links.ts

```diff
@@ -8,203 +8,286 @@
+          ? AppNavLinkStatus.visible
+          : AppNavLinkStatus.hidden,
+      },
+      {
+        id: 'logPatternAnalysis',
+        title: i18n.translate('xpack.ml.deepLink.logPatternAnalysis', {
+          defaultMessage: 'Log Pattern Analysis',
+        }),
+        path: `/${ML_PAGES.AIOPS_LOG_CATEGORIZATION_INDEX_SELECT}`,
+        navLinkStatus: mlCapabilities.canUseAiops
+          ? AppNavLinkStatus.visible
+          : AppNavLinkStatus.hidden,
+      },
+      {
+        id: 'changePointDetections',
+        title: i18n.translate('xpack.ml.deepLink.changePointDetection', {
+          defaultMessage: 'Change Point Detection',
+        }),
+        path: `/${ML_PAGES.AIOPS_CHANGE_POINT_DETECTION_INDEX_SELECT}`,
+        navLinkStatus: mlCapabilities.canUseAiops
+          ? AppNavLinkStatus.visible
+          : AppNavLinkStatus.hidden,
+      },
+    ],
+  };
+}
+
+function getModelManagementDeepLink(mlCapabilities: MlCapabilities): AppDeepLink<LinkId> {
+  return {
+    id: 'modelManagement',
+    title: i18n.translate('xpack.ml.deepLink.modelManagement', {
+      defaultMessage: 'Model Management',
+    }),
+    path: `/${ML_PAGES.TRAINED_MODELS_MANAGE}`,
+    navLinkStatus: mlCapabilities.isNLPEnabled ? AppNavLinkStatus.visible : AppNavLinkStatus.hidden,
+    deepLinks: [
+      {
+        id: 'nodesOverview',
+        title: i18n.translate('xpack.ml.deepLink.trainedModels', {
+          defaultMessage: 'Trained Models',
+        }),
+        path: `/${ML_PAGES.TRAINED_MODELS_MANAGE}`,
+        navLinkStatus: mlCapabilities.isNLPEnabled
+          ? AppNavLinkStatus.visible
+          : AppNavLinkStatus.hidden,
+      },
+      {
+        id: 'nodes',
+        title: i18n.translate('xpack.ml.deepLink.nodes', {
+          defaultMessage: 'Nodes',
+        }),
+        path: `/${ML_PAGES.NODES}`,
+        navLinkStatus:
+          mlCapabilities.isADEnabled && mlCapabilities.isDFAEnabled && mlCapabilities.isNLPEnabled
+            ? AppNavLinkStatus.visible
+            : AppNavLinkStatus.hidden,
+      },
+    ],
+  };
+}
+
+function getMemoryUsageDeepLink(mlCapabilities: MlCapabilities): AppDeepLink<LinkId> {
+  return {
+    id: 'memoryUsage',
+    title: i18n.translate('xpack.ml.deepLink.memoryUsage', {
+      defaultMessage: 'Memory Usage',
+    }),
+    path: `/${ML_PAGES.MEMORY_USAGE}`,
+    navLinkStatus:
+      mlCapabilities.isADEnabled && mlCapabilities.isDFAEnabled && mlCapabilities.isNLPEnabled
+        ? AppNavLinkStatus.visible
+        : AppNavLinkStatus.hidden,
+  };
+}
+
+function getDataVisualizerDeepLink(mlCapabilities: MlCapabilities): AppDeepLink<LinkId> {
+  return {
+    id: 'dataVisualizer',
+    title: i18n.translate('xpack.ml.deepLink.dataVisualizer', {
+      defaultMessage: 'Data Visualizer',
+    }),
+    path: `/${ML_PAGES.DATA_VISUALIZER}`,
+    navLinkStatus:
+      mlCapabilities.isADEnabled && mlCapabilities.isDFAEnabled && mlCapabilities.isNLPEnabled
```

Comment:
do we need all of these for the data visualizer?

Reply:
For now, both data visualizer pages are hidden in serverless ML. This may change in the future if one of the projects decides they want to include them in their nav menu.